README.mkd

```diff
@@ -54,14 +54,40 @@ base and extensions, e.g. `-march=rv64g`. A target `-march` which includes
 floating point instructions implies a hardfloat calling convention, but can be 
 overridden using the `-mabi` flag (see the next section).
 
-The ISA subset naming conventions are described in Chapter 22 of the RISC-V 
-user-level ISA specification. However, tools do not currently follow this 
-specification (no support for parsing version specifiers, input is case 
-sensitive, ...).
+The ISA subset naming conventions and canonical order are described in
+Chapter `ISA Extension Naming Conventions` of the RISC-V user-level ISA
+specification. However, tools can accept the ISA string in non-canonical
+order for reduce the burden of remembering the canonical order.
+
+Detail rule for ISA string:
+
+1. First letter must be `i`, `e` or `g`.
+2. Multi-letter extension must come after single-letter extension.
+3. Single-letter may be non-canonical order.
```

Comment:
we need to consider the single-letter with version like as m1p1,
should it be treated as a single character wholly ?

Reply:
Yeah, I would like to consider single-letter with version as normal single-letter ext. because following reason:
1. ISA manual has example usage like: `RV64I1p0M1p0A1p0F1p0D1p0` and `RV64I1M1A1F1D1`, so I prefer to make those still valid input for toolchain
2. Current arch string parser can handle that, although might be ambiguous when P extension is present.

Anyway, I think we could add note about that :)

Reply:
Sounds good :)